From 9242a1b4be1b8df31a8af91b58a0f53b10cc958e Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Wed, 10 Apr 2024 16:16:14 -0700
Subject: [PATCH] Revert "mm: add per-order mTHP swpin_refault counter"

This reverts commit 37a4ecbf36cbc84bb1aff0f9c81e58261f8826d9.

A temporal fix for
https://lore.kernel.org/all/20240410231538.131175-1-sj@kernel.org/

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/huge_mm.h | 1 -
 mm/huge_memory.c        | 2 --
 mm/memory.c             | 1 -
 3 files changed, 4 deletions(-)

diff --git a/include/linux/huge_mm.h b/include/linux/huge_mm.h
index b67294d5814f..c8256af83e33 100644
--- a/include/linux/huge_mm.h
+++ b/include/linux/huge_mm.h
@@ -269,7 +269,6 @@ enum mthp_stat_item {
 	MTHP_STAT_ANON_ALLOC_FALLBACK,
 	MTHP_STAT_ANON_SWPOUT,
 	MTHP_STAT_ANON_SWPOUT_FALLBACK,
-	MTHP_STAT_ANON_SWPIN_REFAULT,
 	__MTHP_STAT_COUNT
 };
 
diff --git a/mm/huge_memory.c b/mm/huge_memory.c
index 9a1b57ef9c60..68ac27d229ef 100644
--- a/mm/huge_memory.c
+++ b/mm/huge_memory.c
@@ -556,14 +556,12 @@ DEFINE_MTHP_STAT_ATTR(anon_alloc, MTHP_STAT_ANON_ALLOC);
 DEFINE_MTHP_STAT_ATTR(anon_alloc_fallback, MTHP_STAT_ANON_ALLOC_FALLBACK);
 DEFINE_MTHP_STAT_ATTR(anon_swpout, MTHP_STAT_ANON_SWPOUT);
 DEFINE_MTHP_STAT_ATTR(anon_swpout_fallback, MTHP_STAT_ANON_SWPOUT_FALLBACK);
-DEFINE_MTHP_STAT_ATTR(anon_swpin_refault, MTHP_STAT_ANON_SWPIN_REFAULT);
 
 static struct attribute *stats_attrs[] = {
 	&anon_alloc_attr.attr,
 	&anon_alloc_fallback_attr.attr,
 	&anon_swpout_attr.attr,
 	&anon_swpout_fallback_attr.attr,
-	&anon_swpin_refault_attr.attr,
 	NULL,
 };
 
diff --git a/mm/memory.c b/mm/memory.c
index b6fa5146b260..eaface1df978 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -4166,7 +4166,6 @@ vm_fault_t do_swap_page(struct vm_fault *vmf)
 		nr_pages = nr;
 		entry = folio->swap;
 		page = &folio->page;
-		count_mthp_stat(folio_order(folio), MTHP_STAT_ANON_SWPIN_REFAULT);
 	}
 
 check_pte:
-- 
2.39.2

